base/channels.jl

```diff
@@ -472,3 +472,29 @@ function iterate(c::Channel, state=nothing)
 end
 
 IteratorSize(::Type{<:Channel}) = SizeUnknown()
+
+"""
+    Threads.foreach(f, channel::Channel; ntasks=Threads.nthreads())
+
+Similar to `foreach(f, channel)`, but iteration over `channel` and calls to
+`f` are split across `ntasks` Tasks spawned by `Threads.@spawn`. This function
+will wait for all internally spawned tasks to complete before returning.
+"""
+function Threads.foreach(f, channel::Channel; ntasks=Threads.nthreads())
+    stop = Threads.Atomic{Bool}(false)
+    @sync for _ in 1:ntasks
```

Comment:
`@sync` isn't defined yet at the point this is loaded while building the sysimage, which is causing the builds to fail.

Reply:
`@sync` is defined in base/task.jl, which is included directly after base/channels.jl.

Reply:
I naively fixed this by introducing a new file (`threadfuncs.jl`) included after both `channels.jl` and `tasks.jl`. Hopefully isn't too controversial (though the name is bikesheddable, of course 😁).